gcc/rust/ast/rust-ast-full-test.cc

```diff
@@ -4050,6 +4053,82 @@ Module::add_crate_name (std::vector<std::string> &names) const
     item->add_crate_name (names);
 }
 
+static bool
+file_exists (const std::string path)
+{
+  // Simply check if the file exists
+  // FIXME: This does not work on Windows
+  return access (path.c_str (), F_OK) != -1;
+}
+
+// FIXME: This function should also check if the module has a `path` outer
+// attribute and fetch the path from here in that case, i.e:
+// ```
+// #[path="<dir>/<subdir>/<file>.rs"]
```

Comment:
This is a good comment. Lets not fix this within this PR.

But can you investigate how rustc does this? I am interested to know if rustc turns all attributes like this into specific classes or if there are some kind of enum on attributes or something like that to work with them.

Reply:
Definitely! I'd love to work on this